### DISCUSSION TITLE
AWS Lambda run prisma db push get Error: Migration engine exited. Error: Could not find migration-engine binary

### DISCUSSION THREAD

**Discussion by BOBBYUAN** [prisma/prisma#17962] [Q&A] [answered]
I try to run db push prisma through lambda, but get the Error:
```
Datasource "db": PostgreSQL database "*****", schema "public" at "postgrescluster-staging.cluster-c0u6mouxupdz.us-east-1.rds.amazonaws.com:5432"
Error: **Migration engine exited. Error: Could not find migration-engine binary.** Searched in:
- /var/task/node_modules/@prisma/engines/migration-engine-linux-arm64-openssl-1.0.x
- /var/task/node_modules/prisma/migration-engine-linux-arm64-openssl-1.0.x
- /var/task/node_modules/migration-engine-linux-arm64-openssl-1.0.x
- /var/task/node_modules/prisma/runtime/migration-engine-linux-arm64-openssl-1.0.x
```

I tried this solution. [https://github.com/prisma/prisma/issues/17002] and add the command under the build npm install in my project.json

```
 "build-npm-install": {
            "executor": "nx:run-commands",
            "options": {
                "cwd": "dist/infrastructure/data-model-postgres-migrate-handler",
                "color": true,
                "commands": [
                    "mkdir -p node_modules/.prisma",
                    "cp -i -R ../../../node_modules/.prisma node_modules/",
                    "mkdir -p prisma",
                    "cp -i -R ../../../prisma/. prisma/.",
                    "cp prisma/schema.prisma ./",
                    "PRISMA_CLI_BINARY_TARGETS=rhel-openssl-1.0.x npm install --omit=dev",
                    "PRISMA_CLI_BINARY_TARGETS=rhel-openssl-1.0.x npx prisma generate"
                ],
                "parallel": false
            },
            "dependsOn": ["build"]
        },
```

It seems like the lambda is still searching for the wrong migration-engine-linux-arm64 binary, but it should be the rhel-openssl-1.0.x one that works lambda, I have no clue how to fix it in order to run the command **prisma db push in lambda**

Below is how it looks like for the dist folder, which I think the engine is already there

![Screen Shot 2023-02-16 at 1 31 46 PM](https://user-images.githubusercontent.com/25357337/219491509-cb203bc2-73a9-4a75-a7dc-b167b6bb6e8e.png)

**Comment by BOBBYUAN** [comment]
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["postgresqlExtensions"]
  binaryTargets   = ["native", "rhel-openssl-1.0.x"]
}
```

I also setup the right binary target for lambda in the schema.prisma as well

**Reply by nurul3101** [reply]
Are you using the latest prisma version 4.10.1?

**Comment by janpio** [comment]
Weere and how are you actually executing the `db push` command?
How do your deploy your Lambda?
How do you package the content of the archive to be deployed?

**Reply by BOBBYUAN** [reply]
I solved the problem by changing the architecture of the lambda. x86 works for the rhel-openssl-1.0.x. Thanks!

**Comment by BOBBYUAN** [comment, marked as answer]
For anyone has the same problem, rhel-openssl-1.0.x.  works for x86 Lambda, and arm64-openssl-1.0.x works for the arm lambda

**Reply by nurul3101** [reply]
Thanks for sharing the solution that worked for you! 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
